### PR TITLE
Fix login page fragment handling after soft reload on Firefox

### LINKED ISSUE #352: Login page: if reloaded in Firefox, hash will be added again
I've noticed that when (soft-)reloading the login page in Firefox, i.e. simply pressing Ctrl+R (instead of Ctrl+Shift+R for a full reload), there's an annoying behavior on pages which use hashes (fragments) in the URL (i.e. https://example.com/#something). Since Firefox preserves form contents during soft reload (https://bugzilla.mozilla.org/show_bug.cgi?id=46845), the JavaScript embedded in the login page simply adds the hash another time to the hidden `rd` form field.

This causes the user to be redirected to the wrong part of the page after login.

(Using a full reload, i.e. Ctrl+Shift+R, does not have this problem.)

## Expected Behavior

The `rd` field should contain the correct URL so the user is forwarded to the correct part of the site.

## Current Behavior

On every soft reload in Firefox, the hash is added once more, so if the user reloads before proceeding to the login page, she will not return to the correct URL.

## Possible Solution

I guess the best simple way to solve this is to change the script to only append the hash if `inputs[i].value` does not contain `"#"`. Or, alternatively, cut off `/#.*/` before adding `window.location.hash`.

## Steps to Reproduce (for bugs)

See above.

### PR DESCRIPTION
**Pull request by ffdybuster**
## Description

Fixes #352.

## Motivation and Context

Soft reloads on Firefox screw up the redirect link on the sign in page; see #352 for details.

## How Has This Been Tested?

I've downloaded the generated sign in page, manually applied the change from this PR, opened it in Firefox and tested it there (with the developers tools).

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by ffdybuster** [comment]
(Rebased to resolve conflicts.)

**Comment by ffdybuster** [comment]
(Rebased to resolve conflicts.)

File: templates.go [hunk 1]
```diff
@@ -135,6 +135,10 @@ func getTemplates() *template.Template {
 			(function() {
 				var inputs = document.getElementsByName('rd');
 				for (var i = 0; i < inputs.length; i++) {
+					var idx = inputs[i].value.indexOf('#');
+					if (idx >= 0) {
+						inputs[i].value = inputs[i].value.substr(0, idx);
+					}
```
Comment (syscll) [inline templates.go]: Would you mind adding a brief comment to outline what happens here.

Comment (ffdybuster) [reply templates.go]: Sure. I added comments, I hope it's now easier to understand!

**Comment by JoelSpeed** [comment]
Apologies @ffdybuster, because we have recently cut a release, the changelog entry you've added is in the wrong place now, if you could move it up to [the new heading](https://github.com/pusher/oauth2_proxy/blob/master/CHANGELOG.md#changes-since-v500), then I'm happy to get this merged 👍

**Comment by felixfontein** [comment]
@JoelSpeed thanks, I overlooked that it ended up in the wrong place. I've updated the PR.

**Comment by ffdybuster** [comment]
@JoelSpeed @syscll thanks for reviewing this!

**Pull request merged**